server/resp/src/main/java/org/infinispan/server/resp/commands/hash/HDEL.java

```diff
@@ -0,0 +1,51 @@
+package org.infinispan.server.resp.commands.hash;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.multimap.impl.EmbeddedMultimapPairCache;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespErrorUtil;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.util.concurrent.CompletionStages;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * `<code>HDEL key field [field ...]</code>` command.
+ * <p>
+ *    Remove the given fields from the hash stored at key. Unknown fields are just ignored, and a non-existing key
+ *    returns 0.
+ * </p>
+ *
+ * @since 15.0
+ * @see <a href="https://redis.io/commands/hdel/">Redis Documentation</a>
+ * @author José Bolina
+ */
+public class HDEL extends RespCommand implements Resp3Command {
+
+   public HDEL() {
+      super(-3, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler, ChannelHandlerContext ctx, List<byte[]> arguments) {
+      if (arguments.size() < 2) {
+         RespErrorUtil.wrongArgumentNumber(this, handler.allocator());
+         return handler.myStage();
+      }
+
+      EmbeddedMultimapPairCache<byte[], byte[], byte[]> multimap = handler.getHashMapMultimap();
+      CompletionStage<Integer> cs = multimap.remove(arguments.remove(0), arguments.remove(0), arguments.toArray(new byte[0][]));
```

Comment:
```suggestion
      CompletionStage<Integer> cs = multimap.remove(arguments.get(0), arguments.get(1), arguments.subList(2, arguments.size()).toArray(new byte[0][]);
```
Each of these removes would cause a new array allocation.

I would personally recommend to have a different remove method that just takes a Collection of keys instead then we can just pass arguments.subList with minimal allocation required.